yggdrasilffi/src/lib.rs

```diff
@@ -115,6 +112,25 @@ fn result_to_json_ptr<T: Serialize>(result: Result<Option<T>, FFIError>) -> *mut
     CString::new(json_string).unwrap().into_raw()
 }
 
+unsafe fn get_engine(engine_ptr: *mut c_void) -> Result<ManagedEngine, FFIError> {
+    if engine_ptr.is_null() {
+        return Err(FFIError::NullError);
+    }
+    let arc_instance = Arc::from_raw(engine_ptr as *const RawPointerDataType);
+
+    let cloned_arc = arc_instance.clone();
+    forget(arc_instance);
```

Comment:
...but now we have a problem because we've reconstituted the Arc from the raw pointer and when this function exits, it'll be dropped (Crab-Speak for garbage collected).

So we tell the Rust compile to forget about it (please don't clean it up).

This is super important because without this it's possible to hit 0 references to this and have it dropped which means the whole program catches fire